tests/src/test/scala/com/nvidia/spark/rapids/ParquetWriterSuite.scala

```diff
@@ -86,20 +86,26 @@ class ParquetWriterSuite extends SparkQueryCompareTestSuite {
   }
 
   test("sorted partitioned write") {
-    val conf = new SparkConf().set(RapidsConf.SQL_ENABLED.key, "true")
-    val tempFile = File.createTempFile("partitioned", ".parquet")
-    try {
-      SparkSessionHolder.withSparkSession(conf, spark => {
-        import spark.implicits._
-        val df = spark.sparkContext.parallelize((1L to 10000000L))
-            .map{i => ("a", f"$i%010d", i)}.toDF("partkey", "val", "val2")
-        df.repartition(1, $"partkey").sortWithinPartitions($"partkey", $"val", $"val2")
-            .write.mode("overwrite").partitionBy("partkey").parquet(tempFile.getAbsolutePath)
-        val firstRow = spark.read.parquet(tempFile.getAbsolutePath).head
-        assertResult("0000000001")(firstRow.getString(0))
-      })
-    } finally {
-      fullyDelete(tempFile)
+    // Skip this test for Spark 3.4.0+
+    // because there are several related bugs in Spark which haven't been fixed yet:
+    // https://issues.apache.org/jira/browse/SPARK-40588
+    // https://issues.apache.org/jira/browse/SPARK-40885
+    if (cmpSparkVersion(3, 4, 0) < 0) {
```

Comment:
Rather than an explicit `if`, instead this should use something like:

```
assume(!isSpark340OrLater, 
    "Skipping on Spark >= 3.4 due to " +
    "https://issues.apache.org/jira/browse/SPARK-40588 and " +
    "https://issues.apache.org/jira/browse/SPARK-40885")
```

That way we can see in the test logs the test is not actually passed (as the `if` approach would otherwise indicate).